File: Gray-Code/Gray-Code.cpp

class Solution {
public:
    vector<int> grayCode(int n) {
        vector<int> ans;
        ans.push_back(0);
        for(int i=0;i<n;i++)
        {
            int prev=1<<i;
            for(int j=ans.size()-1;j>=0;j--)
                ans.push_back(ans[j]+prev);
        }
        return ans;
    }
};


 

File: solutions/Gray-Code/Gray-Code.py


class Solution:
    # @return a list of integers
    def grayCode(self, n):
        ans = [0]
        for i in range(n):
            left = len(ans)
            for j in range(len(ans)-1,-1,-1):
                ans.append(left | ans[j])
        return ans
 